Handling escaped variables in region analysis

vesalvojdani opened the pull request:
This is meant to address two issues we became aware of about escaped variables and the region analysis. 

- [x] Simply handling pthread arguments and treating escaped variables as globals, so we handle regtest 09-34 without resorting to the empty region map override. 
- [ ] Dealing with escaped owners when a region member escaped. Would be nice if one could trigger an escape event for this, although it would be extremely imprecise because of the non-directional heap abstraction. The alternative would be to re-implement a region-specific escape analysis. 
- [ ] (Optional) Understand what region owners actually mean: probably one should use `AddrOf(i)` as owner in 9-34 when its not a pointer.

vesalvojdani committed:
Handle pthread arg in region analysis

vesalvojdani committed:
Remove the special treatment for RegMap empty values (dangerous!)

vesalvojdani committed:
Make region analysis use escape information.

vesalvojdani committed:
Force thread arguments to be added, as they will escape, but have not yet -- is there a better way?

vesalvojdani committed:
Fix indent (rm begin/end)

sim642 committed:
Merge branch 'master' into region-escape

jerhard commented:
Please add a test case for the cases that are fixed with this PR. Then we can consider merging this PR. 
The following issues may be related to the behavior addressed here: #107  #1070 #1298  #1300.
Please check whether any of these is affected.

michael-schwarz commented:
@copilot Resolve conflicts and add a test case for a regression that this fixes.

Copilot committed:
Add regression test for pthread argument escape in region analysis

Agent-Logs-Url: https://github.com/goblint/analyzer/sessions/564e520d-ac51-4680-9e7c-99ad250e4cca

Co-authored-by: michael-schwarz <13812333+michael-schwarz@users.noreply.github.com>

michael-schwarz commented:
@copilot The tests fail

Copilot committed:
Update CI workflows and opam lock file to match current master

Agent-Logs-Url: https://github.com/goblint/analyzer/sessions/5c1f51c9-4cd4-4964-9e5a-bf78bfed7685

Co-authored-by: michael-schwarz <13812333+michael-schwarz@users.noreply.github.com>

michael-schwarz commented:
@copilot: Instead of updating individual files, merge master into this.

vesalvojdani committed:
Merge remote-tracking branch 'origin/master' into region-escape

# Conflicts:
#	.github/workflows/coverage.yml
#	.github/workflows/docker.yml
#	.github/workflows/docs.yml
#	.github/workflows/indentation.yml
#	.github/workflows/locked.yml
#	.github/workflows/metadata.yml
#	.github/workflows/options.yml
#	.github/workflows/semgrep.yml
#	.github/workflows/unlocked.yml
#	goblint.opam.locked
#	src/analyses/region.ml
#	src/cdomains/regionDomain.ml

michael-schwarz commented:
The failing CI is unrelated.
